crates/iceberg-catalog/src/implementations/postgres/tabular.rs

```diff
@@ -50,11 +50,15 @@ where
         AND n.warehouse_id = $3
         AND w.status = 'active'
         AND t.typ = $4
+        AND (t.deleted_at IS NULL OR $5)
+        AND (t.metadata_location IS NOT NULL OR $6)
```

Comment:
This filter should be removed - the `if staged && !list_flags.include_staged` statement below can now no longer be hit - or we remove the statement below?

Reply:
do you see value in the statement below?

Reply:
It'd probably still have to stay since the metadata_location in the returned struct is optional and we'd like to make sure, right?